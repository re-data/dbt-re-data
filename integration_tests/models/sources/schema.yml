version: 2

sources:
  - name:  "{{ var('source_schema', target.schema) }}"
    tables:
      - name: re_data_source_test_table
        columns:
          - name: number
            tests:
              - not_null
              - unique
          
          - name: description
            tests:
              - not_null
              - unique
          
      