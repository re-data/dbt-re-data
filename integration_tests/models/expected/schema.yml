version: 2

models:
  - name: test_re_data_row_count
    tests:
      - dbt_utils.equality:
          compare_model: ref('expected_sample_data_row_count')
          compare_columns:
            - right(upper(replace(replace(table_name, '"', ''), '`', '')), 10) as table_name
            - metric
            - value
            - time_window_start
            - time_window_end
            - interval_length_sec

  - name: test_re_data_freshness
    tests:
      - dbt_utils.equality:
          compare_model: ref('expected_sample_data_freshness')
          compare_columns:
            - right(upper(replace(replace(table_name, '"', ''), '`', '')), 10) as table_name
            - metric
            - value
            - time_window_end

  - name: test_re_data_max
    tests:
      - dbt_utils.equality:
          compare_model: ref('expected_sample_data_max')
          compare_columns:
            - right(upper(replace(replace(table_name, '"', ''), '`', '')), 10) as table_name
            - coalesce(upper(column_name), '') as column_name
            - metric
            - value
            - time_window_start
            - time_window_end
            - interval_length_sec

  - name: test_re_data_max_length
    tests:
      - dbt_utils.equality:
          compare_model: ref('expected_sample_data_max_length')
          compare_columns:
            - right(upper(replace(replace(table_name, '"', ''), '`', '')), 10) as table_name
            - coalesce(upper(column_name), '') as column_name
            - metric
            - value
            - time_window_start
            - time_window_end
            - interval_length_sec

  - name: test_re_data_z_score
    tests:
      - dbt_utils.equality:
          compare_model: ref('expected_sample_data_z_score')
          compare_columns:
            - right(upper(replace(replace(table_name, '"', ''), '`', '')), 10) as table_name
            - coalesce(upper(column_name), '') as column_name
            - metric
            - cast (z_score_value * 1000 as integer) as z_score_value
            - time_window_end
            - cast (last_value * 1000 as integer) as last_value
            - cast (last_avg * 1000 as integer) as last_avg
            - cast (last_stddev * 1000 as integer) as last_stddev
            - interval_length_sec

  - name: test_re_data_metrics
    tests:
      - dbt_utils.equality:
          compare_model: ref('expected_sample_data_metrics')
          compare_columns:
            - right(upper(replace(replace(table_name, '"', ''), '`', '')), 10) as table_name
            - coalesce(upper(column_name), '') as column_name
            - metric
            - cast (value * 1000 as integer) as value
            - time_window_start
            - time_window_end
            - interval_length_sec

  - name: test_re_data_test_history
    tests:
     - dbt_utils.equality:
         compare_model: ref('expected_sample_data_test_history')
         compare_columns:
          - right(upper(replace(replace(table_name, '"', ''), '`', '')), 10) as table_name
          - coalesce(upper(column_name), '') as column_name
          - right(test_name, 15) as test_name
          - status